ui/dashboards/src/components/GridLayout/GridContainer.tsx

```diff
@@ -0,0 +1,221 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useEffect, useState } from 'react';
+import { styled } from '@mui/material';
+
+export interface GridContainerProps {
+  children: React.ReactNode;
+}
+
+export function GridContainer(props: GridContainerProps) {
+  const [isFirstRender, setIsFirstRender] = useState(true);
+  useEffect(() => {
+    if (isFirstRender) {
+      setIsFirstRender(false);
+    }
+  }, [isFirstRender]);
+
+  return (
+    <ReactGridLayoutContainer
+      sx={{
+        // This adds spcing between grids (rows) in the overall dashboard
+        '& + &': { marginTop: (theme) => theme.spacing(1) },
+        // This disables the animation of grid items when a grid is first rendered
+        // (see https://github.com/react-grid-layout/react-grid-layout/issues/103)
+        '& .react-grid-item.cssTransforms': { transitionProperty: isFirstRender ? 'none' : 'transform' },
+      }}
+    >
+      {props.children}
+    </ReactGridLayoutContainer>
+  );
+}
+
+/**
+ * These are the classes needed by react-grid-layout from their CSS stylesheet.
+ */
+const ReactGridLayoutContainer = styled('section')(({ theme }) => ({
+  '& .react-grid-layout': {
```

Comment:
Out of curiosity, are these classes part of the "public css API" of react-grid-layout?

Reply:
Yup, Eunice basically copied them from the CSS stylesheet that ships with the package a couple months ago.